src/map.cpp

```diff
@@ -2932,15 +2933,24 @@ void map::collapse_at( const tripoint &p, const bool silent )
     }
 }
 
-void map::smash_items( const tripoint &p, const int power )
+void map::smash_items( const tripoint &p, const int power, const std::string &cause_message )
 {
     if( !has_items( p ) ) {
         return;
     }
 
+    // Keep track of how many items have been damaged, and what the most recent one is
+    bool item_was_damaged = false;
+    int items_damaged = 0;
+    std::string most_recently_damaged = "";
```

Comment:
Default constructor creates empty string.

```suggestion
    std::string most_recently_damaged;
```